app/services/update_package_price.rb

```diff
@@ -3,11 +3,22 @@
 class UpdatePackagePrice
   def self.call(package, new_price_cents, **options)
     Package.transaction do
+      price = Price.new(package: package, price_cents: package.price_cents)
+      # Check if municipality option is given or not
+      if options[:municipality]
+        # Add municipality to package
+        municipality = package.municipalities.find_or_initialize_by(name: options[:municipality])
+        municipality.price_cents = new_price_cents
+        municipality.save!
+        # Update price attributes
+        price.municipality = municipality
+        price.price_cents = municipality.price_cents
+      else
+        # Update the current price
+        package.update!(price_cents: new_price_cents)
+      end
       # Add a pricing history record
-      Price.create!(package: package, price_cents: package.price_cents)
-
-      # Update the current price
-      package.update!(price_cents: new_price_cents)
+      price.save!
```

Comment:
Is it correct that running this method we will get a price attached to package and municipality?

Reply:
If we see the backward compatibility then it can generate a price with the only package and with new requirements with the municipality and package also.